renku/core/dataset/dataset.py

```diff
@@ -1315,3 +1321,60 @@ def store_dataset_data_location(dataset: Dataset, location: Optional[Path]) -> N
 def read_dataset_data_location(dataset: Dataset) -> Optional[str]:
     """Read data location for a dataset in the config file."""
     return get_client().get_value(section="dataset-locations", key=dataset.name, config_filter=ConfigFilter.LOCAL_ONLY)
+
+
+@inject.autoparams("client_dispatcher", "storage_factory")
+def mount_external_storage(
+    name: str,
+    existing: Optional[Path],
+    unmount: bool,
+    yes: bool,
+    client_dispatcher: IClientDispatcher,
+    storage_factory: IStorageFactory,
+) -> None:
+    """Mount an external storage to a dataset's data directory.
+
+    Args:
+        name(str): Name of the dataset
+        existing(Optional[Path]): An existing mount point to use instead of actually mounting the external storage.
+        unmount(bool): Unmount the external storage.
+        yes(bool): Don't prompt when removing non-empty dataset's data directory.
```

Comment:
I know it's annoying but to satisfy the pre-commit check (I think `pydocstyle` one), we have to document every argument in the docstring, including the two injected ones. I guess you haven't updated/installed the `pre-commit` config?

I prefer adding everything but then being sure that we don't miss a relevant parameter instead of not checking but not having to document the injected parameters, as it's safer that way.

Reply:
> I guess you haven't updated/installed the pre-commit config?

No,  I don't.